Don't show error in studio when fetching YouTube metadata fails.

mtyaka opened the pull request:
If the YouTube API key was not set correctly, a very confusing error message ("Studio is having trouble saving your work [object Object]") would pop up in the Studio when loading a unit containing at least one YouTube video.

The YouTube API key is not strictly necessary - YouTube videos continue to work correctly without a valid API key. The only thing that the API key is used for is fetching the length of the video prior to playing the video. Without the API key, the duration of the video is displayed as 0:00. If you start playing the video, the duration gets set to the correct value though.

This simple patch suppresses the error message in the Studio and logs a warning to the JS console instead.

![screen shot 2016-09-15 at 12 13 44](https://cloud.githubusercontent.com/assets/32585/18555090/ce0dece0-7b66-11e6-9e74-72b6f287011a.png)

**JIRA tickets**: https://openedx.atlassian.net/browse/OSPR-1450

**Dependencies**: None

**Sandbox URL**: http://studio-pr13475.sandbox.opencraft.hosting/

**Testing instructions**:

1. Open unit containing a YouTube video in the Studio (for example http://studio-pr13475.sandbox.opencraft.hosting/container/block-v1:edX+DemoX+Demo_Course+type@vertical+block@4f6c1b4e316a419ab5b6bf30e6c708e9)
2. Observe that the "Studio is having trouble saving your work [object Object]" is not show, even though the YouTube v3 data API request fails. A warning is logged to the JS console instead:

![screen shot 2016-09-18 at 12 15 46](https://cloud.githubusercontent.com/assets/32585/18614950/b7871b92-7d99-11e6-8436-35cd34a00530.png)

**Reviewers**
- [x] OpenCraft: @itsjeyd
- [ ] edX: @sstack22

edx-webhook commented:
Thanks for the pull request, @mtyaka! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-platform&number=13475

edx-webhook commented:
Thanks for the pull request, @mtyaka! I've created [OSPR-1450](https://openedx.atlassian.net/browse/OSPR-1450) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

bradenmacdonald commented:
@mtyaka Can you please mention in the PR description that the YouTube API key is not actually required, which is why it's OK to suppress this error?

And what do you think about adding an `error` handler to the `$.ajax` call that will log an explanatory message to the JS developer console like "YouTube API request failed - usually this means the YouTube API key is invalid. Some video metadata may be unavailable."

mtyaka commented:
@bradenmacdonald Thanks for the good suggestion! I updated the PR description and added a `$.ajax` error handler that prints an explanatory message to the JS console.

sstack22 commented:
👍  - this looks great, thanks!
